### PR TITLE
Fix ARM64 Docker builds with proper cross-compilation

### LINKED ISSUE #9: Integration Tests
We need integration tests for Openchangelog to ensure key functionality works correctly, especially in edge cases. These tests should run Openchangelog with sqlite (multi-tenant mode) and should use the API client to interact with Openchangelog to get e2e safety.

### LINKED ISSUE #21: Exec error on arm
Hi Jonas,
Awesome work. I was trying to self host this on my raspi 5 and I ran across this error: 
```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
exec /app/openchangelog: exec format error
```
I'm brand new to Docker and docker compose so maybe I am doing something wrong? Or is openchangelog just not supported on arm64

### PR DESCRIPTION
**Pull request by SashankBhamidi**
### What does this PR do?
Fixes ARM Docker build issues and adds comprehensive testing infrastructure along with contribution standardization.

### Why is this change needed?
ARM Docker images were failing with exec format errors on Raspberry Pi devices, and the project lacked integration tests and standardized PR process.

### How was this implemented?
- Added ARM64/ARM v7 cross-compilation toolchain for CGO dependencies
- Fixed conditional build logic for proper architecture targeting
- Created comprehensive integration test suite with 12 test functions
- Added future-proof PR template for standardized contributions

## Related Issues
- Fixes #21 (ARM64 Docker build issues)
- Fixes #9 (Integration tests needed)

## Type of Change
- [x] Bug fix (ARM Docker builds)
- [x] New feature (integration tests, PR template)
- [ ] Breaking change
- [x] Documentation update (PR template)
- [x] Configuration/infrastructure change (GitHub Actions, Dockerfiles)

### How has this been tested?
- Verified cross-compilation generates correct ARM64/ARM v7 binaries
- All 12 integration tests pass (HTTP endpoints, database, RSS, caching, E2E)
- All existing unit tests continue to pass
- Manual testing of build processes

### Test checklist
- [x] Existing tests pass
- [x] Added tests for new functionality (12 integration tests)
- [x] Manually verified changes work as expected
- [x] Tested edge cases and error conditions

## Breaking Changes
None - only adds new functionality and fixes existing issues.

## Dependencies
No new runtime dependencies. Added ARM cross-compilation build tools in Docker images.

## Checklist
- [x] I have read and followed the contributing guidelines
- [x] My code follows the project's coding standards
- [x] I have followed all build and generation steps outlined in CONTRIBUTING.md
- [x] I have updated documentation where necessary (PR template)
- [x] My changes generate no new warnings or errors
- [x] I have considered the impact on existing functionality

## Additional Context
This PR combines multiple improvements that were developed together. The integration tests provide comprehensive coverage for the multi-tenant SQLite functionality, while the ARM fixes ensure broad hardware compatibility. The PR template will help standardize future contributions.

### CODE REVIEW

**Comment by sashankbhamidi** [comment]
Updated to add ARM v7 support for broader Raspberry Pi compatibility:

- ARM64 for Pi 5 and newer models
- ARM v7 for Pi 2, 3, 4 (32-bit OS)
- All architectures tested and working

**Comment by JonasHiltl** [comment]
@sashankbhamidi  Thanks for this PR! Much appreciated. I will have a look at this in the coming days

**Comment by SashankBhamidi** [comment]
Also added a PR template to help standardize future contributions. Figured it would be useful while working on this.

**Comment by SashankBhamidi** [comment]
## Description

### What does this PR do?
Fixes ARM Docker build issues and adds comprehensive testing infrastructure along with contribution standardization.

### Why is this change needed?
ARM Docker images were failing with exec format errors on Raspberry Pi devices, and the project lacked integration tests and standardized PR process.

### How was this implemented?
- Added ARM64/ARM v7 cross-compilation toolchain for CGO dependencies
- Fixed conditional build logic for proper architecture targeting
- Created comprehensive integration test suite with 12 test functions
- Added future-proof PR template for standardized contributions

## Related Issues
- Fixes #21 (ARM64 Docker build issues)
- Fixes #9 (Integration tests needed)

## Type of Change
- [x] Bug fix (ARM Docker builds)
- [x] New feature (integration tests, PR template)
- [ ] Breaking change
- [x] Documentation update (PR template)
- [x] Configuration/infrastructure change (GitHub Actions, Dockerfiles)

## Testing
### How has this been tested?
- Verified cross-compilation generates correct ARM64/ARM v7 binaries
- All 12 integration tests pass (HTTP endpoints, database, RSS, caching, E2E)
- All existing unit tests continue to pass
- Manual testing of build processes

### Test checklist
- [x] Existing tests pass
- [x] Added tests for new functionality (12 integration tests)
- [x] Manually verified changes work as expected
- [x] Tested edge cases and error conditions

## Breaking Changes
None - only adds new functionality and fixes existing issues.

## Dependencies
No new runtime dependencies. Added ARM cross-compilation build tools in Docker images.

## Checklist
- [x] I have read and followed the contributing guidelines
- [x] My code follows the project's coding standards
- [x] I have followed all build and generation steps outlined in CONTRIBUTING.md
- [x] I have updated documentation where necessary (PR template)
- [x] My changes generate no new warnings or errors
- [x] I have considered the impact on existing functionality

## Additional Context
This PR combines multiple improvements that were developed together. The integration tests provide comprehensive coverage for the multi-tenant SQLite functionality, while the ARM fixes ensure broad hardware compatibility. The PR template will help standardize future contributions.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+## Description
```
Comment (JonasHiltl) [inline .github/pull_request_template.md:1]: This whole file is too long and complicated use this:

```md
## Summary
<!-- Briefly describe the purpose of this PR -->

## Changes
<!-- What was done in this PR? -->

## Related
<!-- Issues: "Fixes #123", "Closes #123", "Related to #123" -->

## Checklist
- [ ] I have read and followed the [contributing guidelines](CONTRIBUTING.md)
- [ ] I have updated documentation where necessary
- [ ] Existing tests pass
- [ ] Added tests for new functionality (if applicable)
- [ ] Manually verified changes work as expected
- [ ] I have considered the impact on existing functionality
```

**Review by JonasHiltl — CHANGES_REQUESTED** [review]

**Comment by JonasHiltl** [comment]
@SashankBhamidi Could you add another job to the `test.yaml` GitHub action that runs the integration tests?

Like this:
```yaml
name: test

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  unit-tests:
    runs-on: ubuntu-latest

    steps:
      - name: Set up Go 1.23
        uses: actions/setup-go@v1
        with:
          go-version: 1.23
        id: go

      - name: Check out code
        uses: actions/checkout@v1

      - name: Run Unit Tests
        run: go run gotest.tools/gotestsum@latest --junitfile unit-tests.xml --format pkgname github.com/jonashiltl/openchangelog/...

      - name: Unit Test Summary
        uses: test-summary/action@v2
        with:
          paths: "unit-tests.xml"
        if: always()

  integration-tests:
    runs-on: ubuntu-latest
    needs: unit-tests

    steps:
      - name: Set up Go 1.23
        uses: actions/setup-go@v1
        with:
          go-version: 1.23

      - name: Check out code
        uses: actions/checkout@v1

      - name: Run Integration Tests
        run: go run gotest.tools/gotestsum@latest --junitfile integration-tests.xml -- -v ./integration_test.go

      - name: Integration Test Summary
        uses: test-summary/action@v2
        with:
          paths: "integration-tests.xml"
        if: always()
```

**Review by JonasHiltl — APPROVED** [review]
